pkg/redis/RedisConsumer.php

```diff
@@ -83,15 +110,33 @@ public function reject(Message $message, bool $requeue = false): void
     {
         InvalidMessageException::assertMessageInstanceOf($message, RedisMessage::class);
 
-        // do nothing on reject. redis transport always works in auto ack mode
+        $this->acknowledge($message);
 
         if ($requeue) {
-            $this->context->createProducer()->send($this->queue, $message);
+            $message = RedisMessage::jsonUnserialize($message->getReservedKey());
+            $message->setHeader('attempts', 0);
+
+            if ($message->getTimeToLive()) {
+                $message->setHeader('expires_at', time() + $message->getTimeToLive());
+            }
+
+            $this->getRedis()->lpush($this->queue->getName(), json_encode($message));
         }
     }
 
     private function getRedis(): Redis
     {
         return $this->context->getRedis();
     }
+
+    private function initQueueConsumer(): void
+    {
+        if (null === $this->queueConsumer) {
```

Comment:
if we make a RedisQueueConsumer stateless we could move it to the context and pass to real consumers as a dependency.

Reply:
The init would not be needed.